Runbooks/Reset-AzureNSGRules.ps1

```diff
@@ -0,0 +1,47 @@
+$ErrorActionPreference = 'Stop'
+
+try
+{
+    Write-Output "Logging in to Azure..."
+
+    # Ensures you do not inherit an Azure Context in runbook
+    Disable-AzContextAutosave -Scope Process
+
+    # Connect to Azure with system-assigned managed identity
+    $AzureContext = (Connect-AzAccount -Identity).context
+}
+catch
+{
+    Write-Error -Message $_.Exception
+    throw $_.Exception
+}
+
+if (!($AzureContext.Subscription)) {
+    Write-Error -Message "Managed identity does not have access to any subscriptions"
+}
+
+$SubscriptionId = Get-AutomationVariable -Name 'SubscriptionId'
+$AzureContext = Set-AzContext -SubscriptionId "$SubscriptionId" -DefaultProfile $AzureContext
+
+$NsgReferences = Get-AzResource -Name "*-reference" -ResourceType Microsoft.Network/networkSecurityGroups
+
+if ($NsgReferences) {
+    foreach ($NsgReference in $NsgReferences) {
+        $NsgReference = $NsgReference | Get-AzNetworkSecurityGroup
+
+        $NsgTargetName = [regex]::Matches($NsgReference.Name, "(^\w.+)(-reference)").Groups[1].Value
+        $NsgTarget = Get-AzNetworkSecurityGroup -Name $NsgTargetName
+
+        if ($NsgTarget) {
+            $NsgTarget.SecurityRules = $NsgReference.SecurityRules
+
+            Write-Host "Resetting security rules in '$($NsgTarget.Name)' using '$($NsgReference.Name)' as reference."
+        
+            $NsgTarget | Set-AzNetworkSecurityGroup | Select-Object -ExpandProperty SecurityRules | Format-Table
```

Comment:
Just checking that if one of these fail in the loop, then it will sorta be left in an inconsistent state...
I guess we can just re-run but making sure there are no other implications

Reply:
If it fails then an existing non-standard rule(s) will persist. No other implications.